interface/ceed-vector.c

```diff
@@ -266,6 +265,41 @@ int CeedVectorSyncArray(CeedVector vec, CeedMemType mtype) {
   return 0;
 }
 
+/**
+  @brief Take ownership of the CeedVector array and remove the array from the
+           CeedVector
+
+  @param vec        CeedVector
+  @param mtype      Memory type on which to take the array. If the backend
+                    uses a different memory type, this will perform a copy.
+  @param[out] array Array on memory type mtype, or NULL if array pointer is
+                      not required
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref User
+**/
+int CeedVectorTakeArray(CeedVector vec, CeedMemType mtype, CeedScalar **array) {
+  int ierr;
+
+  if (vec->state % 2 == 1)
+    // LCOV_EXCL_START
+    return CeedError(vec->ceed, 1, "Cannot take CeedVector array, the access "
+                     "lock is already in use");
+  // LCOV_EXCL_STOP
+  if (vec->numreaders > 0)
+    // LCOV_EXCL_START
+    return CeedError(vec->ceed, 1, "Cannot take CeedVector array, a process "
+                     "has read access");
+  // LCOV_EXCL_STOP
```

Comment:
It feels silly to test these when we already test access lock, but I can add a pair of tests for these if desired.